cmd/src/users_clean.go

```diff
@@ -0,0 +1,193 @@
+	}
+	var (
+		daysToDelete       = flagSet.Int("days", 60, "Days threshold on which to remove users, must be 60 days or greater and defaults to this value ")
+		noAdmin            = flagSet.Bool("noAdmin", false, "Omit admin accounts from cleanup")
+		removeNoLastActive = flagSet.Bool("removeNeverActive", false, "removes users with null lastActive value")
+		skipConfirmation   = flagSet.Bool("force", false, "skips user confirmation step allowing programmatic use")
+		apiFlags           = api.NewFlags(flagSet)
+	)
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+		if *daysToDelete < 60 {
+			fmt.Println("-days flag must be set to 60 or greater")
+			return nil
+		}
+
+		ctx := context.Background()
+		client := cfg.apiClient(apiFlags, flagSet.Output())
+
+		query := `
+query Users($first: Int, $query: String) {
+	users(first: $first, query: $query) {
+		nodes {
+			...UserFields
+		}
+	}
+}
+` + userFragment
+
+		// get users to delete
+		var result struct {
+			Users struct {
+				Nodes []User
+			}
+		}
+		if ok, err := client.NewRequest(query, nil).Do(ctx, &result); err != nil || !ok {
+			return err
+		}
+
+		usersToDelete := make([]UserToDelete, 0)
+		for _, user := range result.Users.Nodes {
+			daysSinceLastUse, wasLastActive, err := computeDaysSinceLastUse(user)
+			if err != nil {
+				return err
+			}
+			if !wasLastActive && !*removeNoLastActive {
+				continue
+			}
+			if *noAdmin && user.SiteAdmin {
+				continue
+			}
+			if daysSinceLastUse <= *daysToDelete && wasLastActive {
+				continue
+			}
+			deleteUser := UserToDelete{user, daysSinceLastUse}
+
+			usersToDelete = append(usersToDelete, deleteUser)
+		}
+
+		if *skipConfirmation {
+			for _, user := range usersToDelete {
+				if err := removeUser(user.User, client, ctx); err != nil {
+					return err
+				}
+			}
+			return nil
+		}
+
+		// confirm and remove users
+		if confirmed, _ := confirmUserRemoval(usersToDelete); !confirmed {
+			fmt.Println("Aborting removal")
+			return nil
+		} else {
+			fmt.Println("REMOVING USERS")
+			for _, user := range usersToDelete {
+				if err := removeUser(user.User, client, ctx); err != nil {
+					return err
+				}
+			}
+		}
+
+		return nil
+	}
+
+	// Register the command.
+	usersCommands = append(usersCommands, &command{
+		flagSet:   flagSet,
+		handler:   handler,
+		usageFunc: usageFunc,
+	})
+}
+
+// computes days since last usage from current day and time and UsageStatistics.LastActiveTime, uses time.Parse
+func computeDaysSinceLastUse(user User) (timeDiff int, wasLastActive bool, _ error) {
+	timeNow := time.Now()
+	// handle for null lastActiveTime returned from
+	if user.UsageStatistics.LastActiveTime == "" {
+		wasLastActive = false
+		return 0, wasLastActive, nil
+	}
+	timeLast, err := time.Parse(time.RFC3339, user.UsageStatistics.LastActiveTime)
+	if err != nil {
+		return 0, false, err
+	}
+	timeDiff = int(timeNow.Sub(timeLast).Hours() / 24)
```

Comment:
I prefer to have variable declarations as close to their first use as possible (there's exceptions of course, such as group of declarations, etc.) and in this case I don't think we need the declaration at all.

```suggestion
	// handle for null lastActiveTime returned from
	if user.UsageStatistics.LastActiveTime == "" {
		wasLastActive = false
		return 0, wasLastActive, nil
	}
	lastActive, err := time.Parse(time.RFC3339, user.UsageStatistics.LastActiveTime)
	if err != nil {
		return 0, false, err
	}
	timeDiff = int(time.Since(timeLast).Hours() / 24)
```